R/systemfit-tidiers.R

```diff
@@ -0,0 +1,52 @@
+#' @templateVar class systemfit
+#' @template title_desc_tidy
+#' 
+#' @param x A `systemfit` object produced by a call to [systemfit::systemfit()].
+#' @param conf.int Logical indicating whether or not to include a confidence interval in the tidied output. Defaults to `FALSE`.
+#' @template param_unused_dots
+#' 
+#' @evalRd return_tidy(
+#'   "term", 
+#'   "estimate", 
+#'   "std.error", 
+#'   "p.value",
+#'   "conf.low",
+#'   "conf.high"
+#' )
+#'
+#' @details This tidy method works with any model objects of class `systemfit`. 
+#'          Default returns a tibble of six columns.
+#' @importFrom stats confint
+#' @examples
+#' library(systemfit)
+#' df <- data.frame(X = rnorm(100), Y = rnorm(100), Z = rnorm(100), W = rnorm(100))
+#'
+#` fit <- systemfit(formula = list(Y ~ Z, W ~ X), data = df, method = "SUR")
+#'
+#' tidy(fit)
+#' tidy(fit, conf.int = TRUE)
+#' @export
+#' @seealso [tidy()], [systemfit::systemfit()]
+#' @family systemfit tidiers
+#' @aliases systemfit_tidiers
+tidy.systemfit <- function(x, conf.int = TRUE, ...) {
+  nn <- c("term", "estimate", "std.error", "p.value")
+  
+  sf_summary <- summary(x)
+  sf_coefs <- as.data.frame(sf_summary$coefficients)
+  sf_coefs <- sf_coefs[,- which(names(sf_coefs) == "t value")]
+  sf_coefs <- cbind(term = rownames(sf_coefs), sf_coefs)
+  if(conf.int){
+    nn <- c(nn, "conf.low", "conf.high")
+    sf_cis <- confint(x)
+    sf_cis <- matrix(sf_cis,ncol = 2)
```

Comment:
Just curious -- what happens without this line? Is it necessary to coerce to matrix since we're going to end up as a `tibble` anyway?

Reply:
Without it you get
> Error in as.data.frame.default(x[[i]], optional = TRUE) : 
>  cannot coerce class ‘"confint.systemfit"’ to a data.frame
It's a really weird object and very hard to wrangle:
```r
> str(sf_cis)
 'confint.systemfit' num [1:4, 1:2] -0.0339 -0.2893 -0.2527 -0.186 0.4179 ...
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:4] "eq1_(Intercept)" "eq1_Z" "eq2_(Intercept)" "eq2_X"
  ..$ : chr [1:2] "2.5 %" "97.5 %"
```
LMK if you want to take a different approach

Reply:
No problem, let's stick with this.